planning/motion_velocity_planner/autoware_motion_velocity_road_user_stop_module/src/road_user_stop_module.cpp

```diff
@@ -0,0 +1,1175 @@
+    safety_factor.points = {determined_stop_obstacle->pose.position};
+    safety_factor.is_safe = false;
+    safety_factors.factors.push_back(safety_factor);
+    safety_factors.is_safe = false;
+    safety_factors.header.stamp = clock_->now();
+    safety_factors.header.frame_id = "map";
+  }
+
+  planning_factor_interface_->add(
+    output_traj_points, planner_data->current_odometry.pose.pose, stop_pose,
+    autoware::planning_factor_interface::PlanningFactor::STOP, safety_factors);
+
+  prev_stop_distance_info_ = std::make_pair(output_traj_points, determined_zero_vel_dist.value());
+
+  // verify stop point is on trajectory
+  const auto dist_to_traj =
+    autoware::motion_utils::calcLateralOffset(traj_points, stop_pose.position);
+  if (std::abs(dist_to_traj) > 0.5) {
+    RCLCPP_WARN(
+      logger_, "Stop point lateral offset from trajectory: %.3f m at index %zu", dist_to_traj,
+      *zero_vel_idx);
+  }
+
+  return stop_pose.position;
+}
+
+double RoadUserStopModule::calcDesiredStopMargin(
+  const std::shared_ptr<const PlannerData> planner_data,
+  const std::vector<TrajectoryPoint> & traj_points, const StopObstacle & stop_obstacle,
+  const double dist_to_bumper, const size_t ego_segment_idx,
+  const double dist_to_collide_on_ref_traj) const
+{
+  const auto param = param_listener_->get_params();
+  // calculate default stop margin
+  const double default_stop_margin = [&]() {
+    const double v_ego = planner_data->current_odometry.twist.twist.linear.x;
+    const double v_obs = stop_obstacle.velocity;
+
+    const auto ref_traj_length =
+      autoware::motion_utils::calcSignedArcLength(traj_points, 0, traj_points.size() - 1);
+
+    // For negative velocity obstacles (approaching)
+    if (v_obs < param.stop_planning.max_negative_velocity) {
+      const double a_ego = param.stop_planning.effective_deceleration_opposing_traffic;
+      const double & bumper_to_bumper_distance = stop_obstacle.dist_to_collide_on_decimated_traj;
+
+      const double braking_distance = v_ego * v_ego / (2 * a_ego);
+      const double stopping_time = v_ego / a_ego;
+      const double distance_obs_ego_braking = std::abs(v_obs * stopping_time);
+
+      const double ego_stop_margin = param.stop_planning.stop_margin_opposing_traffic;
+
+      const double rel_vel = v_ego - v_obs;
+      constexpr double epsilon = 1e-6;  // Small threshold for numerical stability
+      if (std::abs(rel_vel) <= epsilon) {
+        RCLCPP_WARN(
+          logger_,
+          "Relative velocity (%.3f) is too close to zero. Using minimum safe value for "
+          "calculation.",
+          rel_vel);
+        return param.stop_planning.stop_margin;
+      }
+
+      const double T_coast = std::max(
+        (bumper_to_bumper_distance - ego_stop_margin - braking_distance +
+         distance_obs_ego_braking) /
+          rel_vel,
+        0.0);
+
+      const double stopping_distance = v_ego * T_coast + braking_distance;
```

Comment:
Should use this function?
https://github.com/autowarefoundation/autoware_core/blob/9549a90814c066823c5b0030231a223051dc1b4b/common/autoware_motion_utils/include/autoware/motion_utils/distance/distance.hpp#L27

Reply:
I found that we can't use above function in this process.